xamarin/Xamarin.Forms
# [Enhancement] Added NavigationPage.IconColorProperty

### Description of Change ###

It makes sense to apply BarTextColor to title AND navigation icon (hamburger, back etc.)


### Issues Resolved ### 

- fixes #5177

### API Changes ###

Added NavigationPage.IconColorProperty

### Platforms Affected ### 

- Android
- iOS

### Behavioral/Visual Changes ###

BarTextColor will manage colors for Title text and Icon

### Before/After Screenshots ### 
Try to set BarTextColor to Red

BEFORE:
<img width="486" alt="2" src="https://user-images.githubusercontent.com/10124814/52520915-5ed58b80-2c80-11e9-9ef0-0e4d37d7a477.png">

AFTER:
<img width="484" alt="1" src="https://user-images.githubusercontent.com/10124814/52520923-6f860180-2c80-11e9-9c27-e915dc2b10ab.png">


### Testing Procedure ###

Set BarTextColor to Red to any navigation page and run issues gallery.
Icon should be Red as well as Title
